extensions/terminal-suggest/src/terminalSuggestMain.test.ts

```diff
@@ -7,36 +7,39 @@ import { deepStrictEqual, strictEqual } from 'assert';
 import 'mocha';
 import { availableSpecs, getCompletionItemsFromSpecs } from './terminalSuggestMain';
 
-const availableCommands = ['cd', 'code', 'code-insiders'];
+const availableCommands = ['cd', 'code', 'code-insiders', 'c++', 'c.test', '.', '..', 'dir', 'ls'];
 const codeOptions = ['-', '--add', '--category', '--diff', '--disable-extension', '--disable-extensions', '--disable-gpu', '--enable-proposed-api', '--extensions-dir', '--goto', '--help', '--inspect-brk-extensions', '--inspect-extensions', '--install-extension', '--list-extensions', '--locale', '--log', '--max-memory', '--merge', '--new-window', '--pre-release', '--prof-startup', '--profile', '--reuse-window', '--show-versions', '--status', '--sync', '--telemetry', '--uninstall-extension', '--user-data-dir', '--verbose', '--version', '--wait', '-a', '-d', '-g', '-h', '-m', '-n', '-r', '-s', '-v', '-w'];
 const localeOptions = ['bg', 'de', 'en', 'es', 'fr', 'hu', 'it', 'ja', 'ko', 'pt-br', 'ru', 'tr', 'zh-CN', 'zh-TW'];
 const categoryOptions = ['azure', 'data science', 'debuggers', 'extension packs', 'education', 'formatters', 'keymaps', 'language packs', 'linters', 'machine learning', 'notebooks', 'programming languages', 'scm providers', 'snippets', 'testing', 'themes', 'visualization', 'other'];
 const logOptions = ['critical', 'error', 'warn', 'info', 'debug', 'trace', 'off'];
 const syncOptions = ['on', 'off'];
+const defaultResourceCompletionItems = ['.', '..'];
 
 const testSpecs: ITestSpec[] = [
-	{ input: '|', expectedCompletionLabels: availableCommands },
-	{ input: 'c|', expectedCompletionLabels: availableCommands },
-	{ input: 'ls && c|', expectedCompletionLabels: availableCommands },
+	{ input: '|', expectedCompletionLabels: availableCommands, resourcesRequested: 'both' },
+	{ input: 'c|', expectedCompletionLabels: availableCommands.filter(c => c.startsWith('c')) },
+	{ input: 'ls && c|', expectedCompletionLabels: availableCommands.filter(c => c.startsWith('c')) },
```

Comment:
I think you'll want to isolate the tests a little more or they'll just get more complicated over time. So instead of having all tests use all completions, have most unit tests focus on specific completion specs, then a small amount that deal with merging and dealing with all of them.

Reply:
makes sense, have just been fixing bugs for now. seems like something to defer.

Reply:
https://github.com/microsoft/vscode/issues/234777